src/EthereumRawTx/Abi/ParamType.php

```diff
@@ -0,0 +1,310 @@
+                $length = Buffer::hex(substr($raw, $startPosition, 64))->getInt() * 2;
+
+                $return = Buffer::hex(substr($raw, $startPosition+64, $length));
+                $position += 64;
+
+                return $return;
+            default:
+                throw new \Exception("Unknown input type {$this->getName()}");
+        }
+    }
+
+    static public function encodeUint(Buffer $value)
+    {
+        if (strlen($value->getHex()) > 64) {
+            // todo check length ?
+            throw new \Exception("TODO");
+        }
+
+        return str_pad($value->getHex(), 64, '0', STR_PAD_LEFT);
+    }
+
+    public function encode($value)
+    {
+        // if is type array
+        if($this->isArray()) {
+            if (!is_array($value)) {
+                throw new \Exception("Value must be an array");
+            }
+
+            if (!$this->isDynamic()) {
+                if (count($value) != $this->getStaticArrayLength()) {
+                    throw new \Exception("Value count does not match expected array length");
+                }
+            }
+
+            $return = [];
+
+            if($this->isDynamic())
+            {
+                $return[] = self::encodeUint(Buffer::int(count($value)));
+            }
+
+            foreach($value as $key => $val) {
+                $return[] = $this->getNestedType()->encode($val);
+            }
+
+            return $return;
+        }
+
+        switch ($this->getName()) {
+            case 'int':
+            case 'uint':
+                if (!$value instanceof Buffer) {
+                    throw new \Exception("Value must be a Buffer");
+                }
+
+                return self::encodeUint($value);
+
+            case 'address':
+                if ($value instanceof Buffer === false) {
+                    throw new \Exception("Value must be an Buffer");
+                }
+                if (strlen($value->getHex()) !== 40) {
+                    throw new \Exception("Address must be 40 chars");
+                }
+
+                return str_pad($value->getHex(), 64, '0', STR_PAD_LEFT);
+
+            case 'bool':
+                if ($value instanceof Buffer === false) {
+                    throw new \Exception("Value must be an Buffer");
+                }
+                return str_pad($value->getHex(), 64, '0', STR_PAD_LEFT);
+
+            case 'string':
+            case 'bytes':
+                if($this->isDynamic()) {
+                    if ($value instanceof Buffer === true) {
+                        throw new \Exception('TODO');
+                        // TODO
+                        //return $this->encodeParam('uint', Buffer::int(strlen($value->getHex()) / 2)).$value->getHex();
+                    }
+                    return StringEncoder::encode($value);
+                }
+
+                if ($value instanceof Buffer === true) {
+                    throw new \Exception('TODO');
```

Comment:
Why Todo ? i would like to send Hexadecimal bytes whithout string encoding

Reply:
todo because i did not write specs. Done now.